client/signup/wpcom-login-form/test/index.jsx

```diff
@@ -12,103 +14,98 @@ describe( 'WpcomLoginForm', () => {
-		expect( form.prop( 'action' ) ).toEqual( 'https://test.wordpress.com/wp-login.php' );
+		const form = container.firstChild;
+		expect( form ).toBeVisible();
+		expect( form ).toHaveAttribute( 'method', 'post' );
+		expect( form ).toHaveAttribute( 'action', 'https://test.wordpress.com/wp-login.php' );
 
 		// form should include default hidden elements
-		expect( wrapper.find( 'form > input[type="hidden"]' ) ).toHaveLength( 4 );
-		expect( wrapper.find( 'form > input[name="log"]' ).prop( 'value' ) ).toEqual( 'log_text' );
-		expect( wrapper.find( 'form > input[name="pwd"]' ).prop( 'value' ) ).toEqual( 'secret' );
-		expect( wrapper.find( 'form > input[name="authorization"]' ).prop( 'value' ) ).toEqual(
+		expect( container.querySelector( 'input[name="log"]' ) ).toHaveValue( 'log_text' );
+		expect( container.querySelector( 'input[name="pwd"]' ) ).toHaveValue( 'secret' );
+		expect( container.querySelector( 'input[name="authorization"]' ) ).toHaveValue(
 			'authorization_token'
 		);
-		expect( wrapper.find( 'form > input[name="redirect_to"]' ).prop( 'value' ) ).toEqual(
+		expect( container.querySelector( 'input[name="redirect_to"]' ) ).toHaveValue(
 			'https://test.wordpress.com'
 		);
 
 		// when update a prop
-		wrapper.setProps( { log: 'another_log' } );
-		expect( wrapper.find( 'form > input[name="log"]' ).prop( 'value' ) ).toEqual( 'another_log' );
+		rerender( <WpcomLoginForm { ...props } log="another_log" /> );
+		const el = container.querySelector( 'input[name="log"]' );
+		expect( el.value ).toEqual( 'another_log' );
 	} );
 
 	test( 'should render extra fields if extraFields prop is passed.', () => {
-		const wrapper = shallow(
-			<WpcomLoginForm
-				{ ...props }
-				extraFields={ {
-					foo: 'bar',
-					lorem: 'ipsum',
-				} }
-			/>,
-			{ disableLifecycleMethods: true }
-		);
+		const extraFields = { foo: 'bar', lorem: 'ipsum' };
+		const { container } = render( <WpcomLoginForm { ...props } extraFields={ extraFields } /> );
 
-		expect( wrapper.find( 'input[type="hidden"]' ) ).toHaveLength( 6 );
-		expect( wrapper.find( 'input[name="foo"]' ).prop( 'value' ) ).toEqual( 'bar' );
-		expect( wrapper.find( 'input[name="lorem"]' ).prop( 'value' ) ).toEqual( 'ipsum' );
+		expect( container.querySelectorAll( 'input[type="hidden"]' ) ).toHaveLength( 6 );
+		expect( container.querySelector( 'input[name="foo"]' ) ).toHaveValue( 'bar' );
+		expect( container.querySelector( 'input[name="lorem"]' ) ).toHaveValue( 'ipsum' );
 	} );
 
 	test( 'its action should be under the wpcom subdomain that `redirectTo` prop contains.', () => {
-		const wrapper = shallow(
-			<WpcomLoginForm { ...props } redirectTo="https://foo.wordpress.com" />,
-			{ disableLifecycleMethods: true }
+		const { container, rerender } = render(
+			<WpcomLoginForm { ...props } redirectTo="https://foo.wordpress.com" />
 		);
 
-		expect( wrapper.find( 'form' ).prop( 'action' ) ).toEqual(
+		expect( container.firstChild ).toHaveAttribute(
```

Comment:
I would have used `screen.getByRole` if it was possible but RTL doesn't find any accessible roles as I stated out in my PR description. I will double-check though to make sure I'm not mixing things up.

Reply:
Oh, I guess we're not providing an accessible name there. As per [docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/form_role) that's a requirement for the `form` role to be implicitly set:

> Note: Using the <form> element will automatically communicate a section of content as a form landmark, if it is provided an accessible name. Developers should always prefer using the correct semantic HTML element over using ARIA.

Is adding an accessible name to that form something we can do? Sounds like a good opportunity for an accessibility improvement. Should be as simple as adding [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) or [`title`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#attr-title).

Reply:
I'm not sure about that. The form is intentionally inaccessible to users so adding a name doesn't provide a lot of value to them. I don't think it does any harm though and it may make the tests more clear (I'm not fully convinced but I don't have a strong opinion).

Reply:
That's a fair reason to not use accessible queries 👍 Let's make sure to add an inline comment in the test and that should suffice.